### PR TITLE
[FIX] fix converter forms to json

### PR DESCRIPTION
**Pull request by tinkoff-dwh**
### What is this PR for?
Fix converter for forms

### What type of PR is it?
[Bug Fix]

### How should this be tested?
jdbc 
1. `select '${checkbox:name=1|2,first|second}'`

### Screenshots (if appropriate)
before
before

after
after

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java [hunk 1]
```diff
@@ -54,7 +55,9 @@
  */
 public class RemoteInterpreter extends Interpreter {
   private static final Logger LOGGER = LoggerFactory.getLogger(RemoteInterpreter.class);
-  private static final Gson gson = new Gson();
+  private static final Gson gson = new GsonBuilder()
+      .registerTypeAdapterFactory(Input.TypeAdapterFactory)
+      .create();
 
 
   private String className;
```
Comment (zjffdu) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java]: Could you just use gui.toJson() in L394 ?

**Comment by tinkoff-dwh** [comment]
@zjffdu @felixcheung 
done

**Comment by felixcheung** [comment]
is test failure related?
```
15:11:43,740  INFO org.apache.zeppelin.spark.SparkInterpreter:979 - Sending metadata to Zeppelin server: {message=Spark UI enabled, url=http://10.20.0.79:4040}
<console>:24: error: not found: value value
       def category(min: Int) = {    if (0 <= value) "error"}
                                              ^
15:11:43,772 DEBUG org.apache.zeppelin.interpreter.Interpreter:197 - key: spark.ui.enabled, value: true
15:11:43,773  INFO org.apache.zeppelin.spark.SparkInterpreter:979 - Sending metadata to Zeppelin server: {message=Spark UI enabled, url=http://10.20.0.79:4040}
defined class Person
15:11:44,112 ERROR org.apache.zeppelin.spark.SparkInterpreter:1256 - 
java.lang.NullPointerException
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:784)
```

**Comment by tinkoff-dwh** [comment]
nope

**Comment by tinkoff-dwh** [comment]
merge?

**Pull request closed without merging**